sync-team/src/github/mod.rs

```diff
@@ -489,90 +516,140 @@ impl SyncGitHub {
         &self,
         expected_repo: &rust_team_data::v1::Repo,
     ) -> anyhow::Result<Vec<EnvironmentDiff>> {
-        let actual_environments = self
+        let mut environment_diffs = Vec::new();
```

Comment:
I see you made some changes on how environments are diffed. Can you extract these changes in a separate PR?